account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -1050,6 +1055,12 @@ def _get_exchange_rate_amount(self, amount, currency_amount, currency, line):
                 self.date,
             )
             to_amount = self.company_id.currency_id.round(to_amount_company_currency)
+        elif self.currency_id == currency and not self.foreign_currency_id:
```

Comment:
To compute the exchange amount, I take the "accounting rate", instead of the odoo rate, even if the foreign currency is not filled (case of journan in foreign currency)